external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java

```diff
@@ -75,20 +75,29 @@ public void prepare(Properties options) {
     public void updateState(List<TridentTuple> tuples, TridentCollector collector) {
         String topic = null;
         try {
-            List<Future<RecordMetadata>> futures = new ArrayList<>(tuples.size());
+            long currentTime=System.currentTimeMillis();
+	     int numberOfRecords = tuples.size();
+	     List<Future<RecordMetadata>> futures = new ArrayList<>(numberOfRecords);
             for (TridentTuple tuple : tuples) {
                 topic = topicSelector.getTopic(tuple);
-
+                Object messageFromTuple = mapper.getMessageFromTuple(tuple);
+		 Object keyFromTuple = mapper.getKeyFromTuple(tuple);
+				
                 if(topic != null) {
-                    Future<RecordMetadata> result = producer.send(new ProducerRecord(topic,
-                            mapper.getKeyFromTuple(tuple), mapper.getMessageFromTuple(tuple)));
-                    futures.add(result);
+                   if(messageFromTuple!=null){
```

Comment:
space between ')' and '{', and left side of '!=' and right side of '!='. may be better to add space between 'f' and '('.